File: dojinyou/3주차/9237.java
import java.io.*;
import java.util.Arrays;

public class Main {

    static class Input {
        int[] dayOfFullyGrownById;

        Input(int[] dayOfFullyGrownById) {
            this.dayOfFullyGrownById = dayOfFullyGrownById;
        }
    }

    static class Output {
        int maxDays;

        Output(int maxDays) {
            this.maxDays = maxDays;
        }
    }

    public static void main(String[] args) throws IOException {
        Input input = getInput();
        Output output = solve(input);
        resolve(output);
    }

    private static Input getInput() throws IOException {
        Input input;
        try (BufferedReader br = new BufferedReader(new InputStreamReader(System.in))) {
            br.readLine();
            int[] dayOfFullyGrownById = Arrays.stream(br.readLine().split(" ")).mapToInt(Integer::parseInt).toArray();
            input = new Input(dayOfFullyGrownById);
        }
        return input;
    }

    private static void resolve(Output output) throws IOException {
        try (BufferedWriter bw = new BufferedWriter(new OutputStreamWriter(System.out))) {
            bw.write(String.valueOf(output.maxDays));
        }
    }

    private static Output solve(Input input) {
        int[] sortedDayOfFullyGrown = Arrays.stream(input.dayOfFullyGrownById).sorted().toArray();

        int maxDays = 0;
        for (int i = 1; i <= sortedDayOfFullyGrown.length; i++) {
            int dayOfFullyGrown = sortedDayOfFullyGrown[sortedDayOfFullyGrown.length - i];
            maxDays = Math.max(maxDays, dayOfFullyGrown + i);
        }

        return new Output(maxDays + 1);
    }

}

File: keemsebin/3주차/9237.js
const readline = require("readline");
const rl = readline.createInterface({
  input: process.stdin,
  output: process.stdout,
});

let input = [];

rl.on("line", function (line) {
  input.push(line);
}).on("close", function () {
  let N = Number(input[0]);
  let arr = input[1].split(" ").map(Number);

  arr.sort((a, b) => b - a);
  let max = 0;
  for (let i = 0; i < N; i++) {
    arr[i] += i + 1; // 나무가 다 자란 다음날 이장님 초대
    max = Math.max(max, arr[i]);
  }
  console.log(max + 1); // 1을 더해주는 이유는 0부터 시작했기 때문
});
